distributed/client.py

```diff
@@ -2609,9 +2609,15 @@ def get(
         --------
         Client.compute: Compute asynchronous collections
         """
+        keys_flatten = set(flatten([keys]))
+
+        for key in keys_flatten:
+            if key not in dsk:
+                raise KeyError("Key '%s' not found in the dask graph" % key)
```

Comment:
Can you check the performance of this vs. 

```python
if keys_flatten - set(dsk):
```

I suppose the performance will depend on the relative  sizes of `keys_flatten` and `dsk`.

Reply:
I get the following metrics:
```
def flatten(seq, container=list):
    if isinstance(seq, str):
        yield seq
    else:
        for item in seq:
            if isinstance(item, container):
                for item2 in flatten(item, container=container):
                    yield item2
            else:
                yield item

dsk = {"x": (op.add, 1, 2)}

keys = ["x", "y"]
                
keys_flatten = set(flatten([keys]))
```
```
def f1():
    for key in keys_flatten:
        if key not in dsk:
#             raise KeyError("Key '%s' not found in the dask graph" % key)
            pass
            
%timeit f1()
```
`147 ns ± 1.86 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)`

```
def f2():
    not_found = keys_flatten - set(dsk)
#     raise KeyError("Key '%s' not found in the dask graph" % key)
    pass
    
%timeit f2()
```
`230 ns ± 8.82 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
​`

Reply:
Thanks. I'm more concerned when `dsk` is very large.

It seems like this might fit a bit better in `_graph_to_futures`, where we already iterate over `dsk` a few times, and already have the set of keys (called `keyset`). Can you see if there's a good spot somewhere there?

Reply:
This [line](https://github.com/dask/distributed/blob/2d54ef947a7fe372b1fafc40d2f2b89ff88e9449/distributed/client.py#L2496) could be a spot, but its not possible because that defeats the purpose of using `futures`. And a key can exist in dsk but not in keyset.
Another spot is [line](https://github.com/dask/distributed/blob/2d54ef947a7fe372b1fafc40d2f2b89ff88e9449/distributed/client.py#L2544).

This line could be moved to the top of `_graph_to_futures` function body:
```
futures = {}
for key in keyset:
            if key in dsk:
                futures[key] = Future(key, self, inform=False)
            else:
                raise KeyError("Key '%s' not found in the dask graph" % key)
```

Reply:
Adding the lines to `_graph_to_futures` breaks `client#submit` calls.